rust/src/message/sum2.rs

```diff
@@ -25,14 +59,22 @@ impl Sum2MessageBuffer<Vec<u8>> {
             bytes: vec![0_u8; len],
         }
     }
+}
 
-    /// Create a sum2 message buffer from `bytes`. Fails if the `bytes` don't conform to the
-    /// expected sum2 message length `exp_len`.
-    fn try_from(bytes: Vec<u8>, exp_len: usize) -> Result<Self, PetError> {
-        if bytes.len() != exp_len {
-            return Err(PetError::InvalidMessage);
+impl TryFrom<Vec<u8>> for Sum2MessageBuffer<Vec<u8>> {
```

Comment:
Same question than above: couldn't this be implemented for any `T: AsRef<[u8]>`?